### PR TITLE
Fix the issue when using cvc5 as backend.

### PR DESCRIPTION
**Pull request by Ying1123**
This pr change the` Logic = "ALL_SUPPORTED";` to `Logic = "ALL";` which is consistent with SMT-LIB standard.
Starting from CVC4 1.8 https://github.com/cvc5/cvc5/pull/6664, CVC4/CVC5 no longer support the syntax ALL_SUPPORTED.

### CODE REVIEW

File: Source/Provers/SMTLib/SMTLibProverOptions.cs [hunk 1]
```diff
@@ -139,7 +139,7 @@ public override void PostParse()
         case SolverKind.CVC4:
           SolverArguments.Add(
             "--lang=smt --no-strict-parsing --no-condense-function-values --incremental --produce-models");
-          if (Logic == null) Logic = "ALL_SUPPORTED";
+          if (Logic == null) Logic = "ALL";
```
Comment (shazqadeer) [inline Source/Provers/SMTLib/SMTLibProverOptions.cs:142]: @Ying1123 :  This looks great.  Could you please add a comment about the version of CVC5 which is needed for this change. I am also wondering if we should either 
(1) rename SolverKind.CVC4 to SolverKind.CVC5, or
(2) add a different option called SolverKind.CVC5 keeping the original option unchanged.

Do you have any advice?

@bkragl, @RustanLeino , @zvonimir  :  FYI

Comment (Ying1123) [reply Source/Provers/SMTLib/SMTLibProverOptions.cs:142]: I have added.

Comment (shazqadeer) [reply Source/Provers/SMTLib/SMTLibProverOptions.cs:142]: @Ying1123:   Is there any reason to support both CVC4 and CVC5 in Boogie?  If note, I recommend renaming SolverKind.CVC4 to SolverKind.CVC5 and the command-line option string to "cvc5". If you search for CVC4 and "cvc4" in the Boogie code, you should be able to find all occurrences.

Comment (Ying1123) [reply Source/Provers/SMTLib/SMTLibProverOptions.cs:142]: I think there is no reason to have both. And I can change it into cvc5.

File: Source/Provers/SMTLib/SMTLibProverOptions.cs [hunk 2]
```diff
@@ -166,8 +166,8 @@ public override string Help
           @"
 SMT-specific options:
 ~~~~~~~~~~~~~~~~~~~~~
-SOLVER=<string>           Use the given SMT solver (z3, cvc4, yices2; default: z3)
-LOGIC=<string>            Pass (set-logic <string>) to the prover (default: empty, 'ALL_SUPPORTED' for CVC4)
+SOLVER=<string>           Use the given SMT solver (z3, cvc5, yices2; default: z3)
+LOGIC=<string>            Pass (set-logic <string>) to the prover (default: empty, 'ALL_SUPPORTED' for CVC5)
```
Comment (shazqadeer) [inline Source/Provers/SMTLib/SMTLibProverOptions.cs:170]: ALL_SUPPORTED -> ALL

**Review by shazqadeer — APPROVED** [review]

**Pull request merged**